--- zfs-fuse-0.7.2.2/src/zfs-fuse/zfs_operations.c.orig	2023-12-29 15:58:42.733647041 +0100
+++ zfs-fuse-0.7.2.2/src/zfs-fuse/zfs_operations.c	2023-12-29 16:25:28.658280334 +0100
@@ -34,7 +34,7 @@
 #include <sys/zfs_vfsops.h>
 #include <sys/zfs_znode.h>
 #include <sys/mode.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <attr/attributes.h>
 #include <sys/fcntl.h>
 #include <sys/sa.h>
